### PR TITLE
Visual studio: Generate vcxproj.filters files to adds filters to imit…

### PR DESCRIPTION
**Pull request by rlavarec-gpsw**
…ated directories to navigate more easily in the source files.

### CODE REVIEW

File: mesonbuild/backend/vs2010backend.py [hunk 1]
```diff
@@ -1437,6 +1489,12 @@ def path_normalize_add(path, lis):
                     self.add_include_dirs(lang, inc_cl, inc_dirs)
                     # XXX: Do we need to set the object file name name here too?
 
+        # Filter information
+        filter_group = ET.SubElement(root_filter, 'ItemGroup')
+        for filter_dir in list_filters_path:
+            filter = ET.SubElement(filter_group, 'Filter', Include=filter_dir)
+            ET.SubElement(filter, 'UniqueIdentifier').text = '{' + str(uuid.uuid1()) + '}'
```
Comment (bruchar1) [inline mesonbuild/backend/vs2010backend.py:1496]: I think you should convert the uuid in uppercase. Also, would `uuid4` be better than `uuid1`? Maybe it is not relevant in this context...

Comment (rlavarec-gpsw) [reply mesonbuild/backend/vs2010backend.py:1496]: Hello Charles,
Thanks for the quick review.
Default uuid are generated in uppercase when you create a project from scratch but, uuid are in lowercase if you create manually a filter inside Visual Studio.
I think it doesn't matter much for Visual Studio.

Yes uuid4 seems to be more appropriate than uuid1. Thanks.
I'll make a fix.

Comment (rlavarec-gpsw) [reply mesonbuild/backend/vs2010backend.py]: If you want I'll put in uppercase the uuid. (It is working without)

**Comment by jpakkane** [comment]
As this is new functionality please add a release note snippet explaining what this does and how one should use it.

**Comment by rlavarec-gpsw** [comment]
This is very basic, it automatically generates some vcxproj.filters without options. The filters will try to reflect the structural hierarchy of directories and files.

**Comment by rlavarec-gpsw** [comment]
Do I have to create Release-notes-for-0.65.0.md ?

**Comment by jpakkane** [comment]
No, a snippet file in the `snippets` directory. The final document is autogenerated from those.

**Comment by rlavarec-gpsw** [comment]
Hello,
@nirbheek, @jpakkane
Could this PR be included inside 1.1.1 please.
Thanks.

**Comment by eli-schwartz** [comment]
No, it is a new feature and not a bug fix.

**Pull request merged**